sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsModelFactory.cs

```diff
@@ -556,5 +544,246 @@ public static RecognizeLinkedEntitiesResultCollection RecognizeLinkedEntitiesRes
+            DateTimeOffset completedOn)
+        {
+            return new RecognizeLinkedEntitiesActionResult(result, completedOn);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="TextAnalytics.RecognizeLinkedEntitiesActionResult"/> for mocking purposes.
+        /// </summary>
+        /// <param name="code">Sets the <see cref="TextAnalyticsError.ErrorCode"/> property.</param>
+        /// <param name="message">Sets the <see cref="TextAnalyticsError.Message"/> property.</param>
+        /// <returns>A new instance of <see cref="TextAnalytics.RecognizeLinkedEntitiesActionResult"/> for mocking purposes.</returns>
+        public static RecognizeLinkedEntitiesActionResult RecognizeLinkedEntitiesActionResult(
+            string code,
+            string message)
+        {
+            return new RecognizeLinkedEntitiesActionResult(new TextAnalyticsErrorInternal(code, message));
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="TextAnalytics.RecognizeEntitiesActionResult"/> for mocking purposes.
+        /// </summary>
+        /// <param name="result">Sets the <see cref="RecognizeEntitiesActionResult.Result"/> property.</param>
+        /// <param name="completedOn">Sets the <see cref="TextAnalyticsActionResult.CompletedOn"/> property.</param>
+        /// <returns>A new instance of <see cref="TextAnalytics.RecognizeEntitiesActionResult"/> for mocking purposes.</returns>
+        public static RecognizeEntitiesActionResult RecognizeEntitiesActionResult(
+            RecognizeEntitiesResultCollection result,
+            DateTimeOffset completedOn)
+        {
+            return new RecognizeEntitiesActionResult(result, completedOn);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="TextAnalytics.RecognizeEntitiesActionResult"/> for mocking purposes.
+        /// </summary>
+        /// <param name="code">Sets the <see cref="TextAnalyticsError.ErrorCode"/> property.</param>
+        /// <param name="message">Sets the <see cref="TextAnalyticsError.Message"/> property.</param>
+        /// <returns>A new instance of <see cref="TextAnalytics.RecognizeEntitiesActionResult"/> for mocking purposes.</returns>
+        public static RecognizeEntitiesActionResult RecognizeEntitiesActionResult(
+            string code,
+            string message)
+        {
+            return new RecognizeEntitiesActionResult(new TextAnalyticsErrorInternal(code, message));
+        }
+
+        #endregion Action Result Models
+
+        #region Healthcare
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="TextAnalytics.AnalyzeHealthcareEntitiesResult"/> for mocking purposes.
+        /// </summary>
+        /// <param name="id">Sets the <see cref="TextAnalyticsResult.Id"/> property.</param>
+        /// <param name="statistics">Sets the <see cref="TextAnalyticsResult.Statistics"/> property.</param>
```

Comment:
should this be `AnalyzeHealthcareEntitiesResult .Statistics`?

Reply:
@maririos 
not actually!
i ran into this problem multiple times
where the 'property' i'm trying to reference isn't part of the model we're trying to create, but actually a parent or sometime a parent of a parent class
so what do you think we should reference in this case?

Reply:
Thanks for the clarification. I see we maintain the pattern you are using in other methods so we can leave as is then

Reply:
in this case **Statistics** is part of **TextAnalyticsResult** and not **AnalyzeHealthcareEntitiesResult**
so when trying to reference it as part of **AnalyzeHealthcareEntitiesResult**, the compiler says undefined